src/plugins/profiling/server/routes/search_topn.ts

```diff
@@ -6,11 +6,68 @@
  * Side Public License, v 1.
  */
 import { schema } from '@kbn/config-schema';
-import type { IRouter } from 'kibana/server';
-import { IEsSearchRequest } from '../../../data/server';
-import { IEsSearchResponse } from '../../../data/common';
+import type { IRouter, KibanaResponseFactory } from 'kibana/server';
+import {
+  AggregationsHistogramBucket,
+  AggregationsMultiBucketAggregateBase,
+} from '@elastic/elasticsearch/lib/api/types';
 import type { DataRequestHandlerContext } from '../../../data/server';
 import { getRemoteRoutePaths } from '../../common';
+import { projectTimeRangeQuery, autoHistogramSumCountOnGroupByField } from './mappings';
+
+export async function topNElasticSearchQuery(
+  context: DataRequestHandlerContext,
+  index: string,
+  projectID: string,
+  timeFrom: string,
+  timeTo: string,
+  searchField: string,
+  response: KibanaResponseFactory
+) {
+  const esClient = context.core.elasticsearch.client.asCurrentUser;
+  const resTopNStackTraces = await esClient.search(
+    {
+      index,
+      body: {
+        query: projectTimeRangeQuery(projectID, timeFrom, timeTo),
+        aggs: {
+          histogram: autoHistogramSumCountOnGroupByField(searchField),
+        },
+      },
+    },
+    {}
+  );
+
+  if (searchField === 'StackTraceID') {
+    const docIDs: string[] = [];
```

Comment:
Constructs like this may eat up CPU (when resizing while pushing) and also create GC workload. We may want to think about a reasonable array size to begin with.